src/target.zig

```diff
@@ -487,6 +487,35 @@ pub fn hasRedZone(target: std.Target) bool {
 pub fn libcFullLinkFlags(target: std.Target) []const []const u8 {
     // The linking order of these is significant and should match the order other
     // c compilers such as gcc or clang use.
+
+    // Starting with glibc 2.28, we need to link "c_nonshared" for "pthread_atfork" function
+    // Previously linking "pthread" was sufficient
+    // See https://sourceware.org/git/?p=glibc.git;a=commit;h=1a2f44a848663036c8a14671fe0faa3fed0b2a25
+    const link_c_nonshared = blk: {
+        if (target.isGnuLibC()) {
+            const ver = target.os.version_range.linux.glibc;
+            const order = ver.order(.{ .major = 2, .minor = 28 });
+            break :blk switch (order) {
+                .gt, .eq => true,
+                .lt => false,
+            };
+        } else {
+            break :blk false;
+        }
+    };
+
+    if (link_c_nonshared) {
+        return &[_][]const u8{
+            "-lm",
+            "-lpthread",
+            "-lc",
+            "-lc_nonshared",
+            "-ldl",
+            "-lrt",
+            "-lutil",
+        };
+    }
```

Comment:
Glibc is only present on linux. I think all this code should be moved into a linux branch of the switch below.

Reply:
ok

Reply:
> Glibc is only present on linux.

And GNU/Hurd.

Reply:
I'm not sure if anyone has ever tried compiling zig code targeting hurd, though it seems we do have a `hurd` target os.

Anyhow, I suspect there will be many other issues to cross before we could even test if this works on hurd.